### ISSUE TITLE
Table component: white shadow persist when scroll back to the left

### ISSUE THREAD

**Issue by ecgan** [woothemes/woocommerce#52831]
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

In the WooCommerce Table component (which is used in Analytics reports such as Analytics > Orders page at `/wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Forders`), when the table is scrollable, there will be white shadow on the left and right. When we scroll all the way to the end, the white shadow at that end should disappear.

The issue here is that when we scroll to the right and then back to the left all the way, the white shadow on the left persist and does not disappear.

Demo video:

https://github.com/user-attachments/assets/958d1c09-daf0-4dd2-a951-9b55b79eed35

### Expected behavior

When we scroll to the right and then back to the left all the way, the white shadow should disappear.

### Actual behavior

When we scroll to the right and then back to the left all the way, the white shadow on the left persist and does not disappear.

### Steps to reproduce

1. Go to Analytics > Orders page: `/wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Forders`
2. Go to the Orders table and scroll horizontally in the table.

### WordPress Environment

Not relevant.

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by aheckler** [comment]
A perhaps related issue @joelclimbsthings: if you shrink the viewport size down so that the gradient/fade effect is applied, it's not un-applied when you widen the viewport again. 

https://github.com/user-attachments/assets/7677f7fc-bfa5-4d24-8f07-32ee248ba848

**Comment by ecgan** [comment]
@aheckler , thanks for the demo video. I can confirm that I can reproduce the issue using the latest develop branch (containing @joelclimbsthings's PR https://github.com/woocommerce/woocommerce/pull/52795). I'm reopening this issue for now. We can consider creating a separate issue if we want.

**Comment by ecgan** [comment]
I looked into the code, and it seems like there are issues with the states and effects not firing correctly.

Ideally, the states (`isScrollableRight`) should correspond to the classes applied to the container div with class name `woocommerce-table__table is-scrollable-right`. However, when we are resizing the viewport, we would notice that the div has the class `is-scrollable-right` (causing the white shadow to be displayed on the right), but the state `isScrollableRight` is false (it should be true when the white shadow is shown). This causes the white shadow not to be toggled off. The relevant code is here:

https://github.com/woocommerce/woocommerce/blob/855111f0620e45858ba2760e646afc463c4c1ab4/packages/js/components/src/table/table.tsx#L135-L154

I noticed there are some `eslint[react-hooks/exhaustive-deps]` issues in the `useEffect` hook usage in the `table.tsx` file. I suspect this is one cause of the issue. We may want to fix the eslint issue to ensure things work correctly.

**Comment by ecgan** [comment]
Adding one idea that I found: we can remove the table shadow completely when the table `clientWidth` is the same as `scrollWidth`. 

Source: https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollWidth

> If the element's content can fit without a need for horizontal scrollbar, its `scrollWidth` is equal to [`clientWidth`](https://developer.mozilla.org/en-US/docs/Web/API/Element/clientWidth).

**Comment by joelclimbsthings** [comment]
Thanks @aheckler and @ecgan !

I've been able to reproduce this as well. That said, it's a fairly niche/minor issue at this point, and given that we're about to get to work on replacing all of this, it may not be worth spending too much time on.

I did look briefly at this and this line:
```js
const scrolledToEnd =
				table.scrollWidth - table.scrollLeft <= table.offsetWidth
```
The above is just a pixel or two off in my tests to correctly disable the box shadow. We could tweak this, but again, I'm not sure I'd want to spend much time on a more thorough fix. What do you guys think?

**Comment by ecgan** [comment]
> We could tweak this, but again, I'm not sure I'd want to spend much time on a more thorough fix. What do you guys think?

@joelclimbsthings , yeah I agree with you too. It's a niche issue. I would treat it as low priority. Maybe we can tackle this issue when we have the capacity or during porter rotation in the future.
